client/src/pages/SingleStar.js

```diff
@@ -9,101 +9,100 @@ import ListGroup from "react-bootstrap/ListGroup";
-										<p>
-											{comment.commenter} - {comment.datetime}
-										</p>
-										<p>{comment.text}</p>
-									</ListGroup.Item>
-								))}
-							</ListGroup>
-						)}
-					</Col>
-					<Col
-						sm={12}
-						md={6}
-						className="d-flex justify-content-center align-items-center"
-					>
-						<Button onClick={() => setEditing(true)} className="ml-">
-							Edit STAR
-						</Button>
-						<EditForm
-							active={editing}
-							star={star}
-							refreshStars={refreshStars}
-						/>
-					</Col>
-				</Row>
-			</Card.Body>
-		</Card>
-	);
+    const [editing, setEditing] = useState(false);
+    const [showCommentModal, setShowCommentModal] = useState(false);
+    const refreshStars = (refresh) => {
+        if (refresh) {
+            setStars(null);
+        }
+        setEditing(null);
+    };
+    // Helper function to handle adding a new comment
+    const handleAddComment = (comment) => {
+        // Add the comment to the star object
+        star.comment.push(comment);
+        // Update the list of stars
+        setStars(null);
+        // Hide the comment modal
+        setShowCommentModal(false);
+    };
+    return (
+        <Card key={star.id} className="my-3 shadow">
+            <Card.Body>
+                <Row>
+                    <Col sm={12} md={6}>
+                        <Card.Title>{star.name}</Card.Title>
+                        <Card.Text>
+                            <span style={{ fontSize: "1rem", fontWeight: "bold" }}>
+                                Situation:
+                            </span>{" "}
+                            {star.situation}
+                        </Card.Text>
+                        <Card.Text>
+                            <span style={{ fontSize: "1rem", fontWeight: "bold" }}>
+                                Task:
+                            </span>{" "}
+                            {star.task}
+                        </Card.Text>
+                        <Card.Text>
+                            <span style={{ fontSize: "1rem", fontWeight: "bold" }}>
+                                Action:
+                            </span>{" "}
+                            {star.action}
+                        </Card.Text>
+                        <Card.Text>
+                            <span style={{ fontSize: "1rem", fontWeight: "bold" }}>
+                                Result:
+                            </span>{" "}
+                            {star.result}
+                        </Card.Text>
+                        <Card.Text>
+                            <span style={{ fontSize: "1rem", fontWeight: "bold" }}>
+                                Description:
+                            </span>{" "}
+                            {star.description}
+                        </Card.Text>
+                        <div>
+                        <Badge variant="secondary">{star.comment && star.comment.length}</Badge>{" "}
```

Comment:
Hi Fatemeh, i don't think you need to implement the comment. length on the front end in order to access the number of comments left. We have already done it on the back end, thus once the stars are fetched from the back end, the number of comments will be included.  All we need to do is add a badge to it. You can refer to what I have done on a different branch

Reply:
Hey Tsino, thank you for your review.I removed it.  I approved issue 72 today.